src/map/skill.c

```diff
@@ -15623,20 +15623,49 @@ bool skill_check_condition_castbegin(struct map_session_data* sd, uint16 skill_i
 		}
 	}
 
-	//check if equipped item
+	// Check for equipped item(s)
 	if (require.eqItem_count) {
+		uint8 count = require.eqItem_count;
+
 		for (i = 0; i < require.eqItem_count; i++) {
 			uint16 reqeqit = require.eqItem[i];
 
 			if (!reqeqit)
-				break; //no required item; get out of here
-			if (!pc_checkequip2(sd,reqeqit,EQI_ACC_L,EQI_MAX)) {
-				if (i == require.eqItem_count) {
-					clif_skill_fail(sd,skill_id,USESKILL_FAIL_THIS_WEAPON,0);
-					return false;
-				}
-			} else
-				break; // Wearing an applicable item.
+				break; // Skill has no required item(s); get out of here
+			switch(skill_id) { // Specific skills require multiple items while default will handle singular cases
+				case NC_PILEBUNKER:
+					if (!pc_checkequip2(sd,reqeqit,EQI_ACC_L,EQI_MAX)) {
+						count--;
+						if (!count) {
```

Comment:
Since up at the top you have !reqeqit this can cause problems if there is a zero in between valid items.
Nevertheless you could get rid of the counting and use something like this:
`if( ( i + 1 ) < require.eqItem_count && !require.eqItem[i+1] )`
With this you can peek the next entry and display the error message in that case.
Would have the same problem with zeros in between though.

Reply:
During the parsing process of `skill_require_db` we check whether these item ID are valid or not. So we shouldn't have any problems at this point.